sbomtool/cmd/sbomtool/main.go

```diff
@@ -115,14 +465,79 @@ func runGenerate(cmd *cobra.Command, args []string) error {
 		"spdx", spdx,
 		"cyclonedx", cyclonedx)
 
+	slog.Info("Starting SBOM generation",
+		"package_name", name,
+		"build_directory", buildDir)
+
+	var formats []string
+	if spdx {
+		formats = append(formats, "SPDX 2.3")
+	}
+	if cyclonedx {
+		formats = append(formats, "CycloneDX 1.6")
+	}
+	slog.Info("Generating SBOM formats", "formats", formats)
+
+	if err := performPreflightChecks(buildDir, outDir); err != nil {
```

Comment:
It seems like `performPreflightChecks` performs some duplicate validations that are individually implemented in the next lines.